spec/models/project/storage_spec.rb

```diff
@@ -0,0 +1,114 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe Project::Storage, type: :model do
+  let(:project1) { FactoryGirl.create(:project).reload }
+
+  let(:project2) { FactoryGirl.create(:project) }
+  let(:repository2) { FactoryGirl.create(:repository_git, project: project2) }
+
+  let(:wp) { FactoryGirl.create(:work_package, project: project1) }
+  let(:wikipage) {
+    FactoryGirl.create(:wiki_page,
+                       wiki: project1.wiki)
+  }
+
+  before do
+    FactoryGirl.create_list(:attachment, 10, filesize: 250, container: wp)
+    FactoryGirl.create(:attachment, filesize: 10000, container: wikipage)
+
+    repository2.update_attributes(required_storage_bytes: 1234)
+  end
+
+  describe '#with_required_storage' do
+    it 'counts projects correctly' do
+      p1, p2 = Project.with_required_storage.find(project1.id, project2.id)
+
+      # The automatically created wiki for a project does not seem to
+      # be properly reloaded here in all cases.
+      project1.wiki.reload
+
+      expect(p1.work_package_required_space).to eq('2500')
```

Comment:
Yeah, mainly running pg locally and was waiting for travis this time. I'm usually running mysql on a VM, too, but that currently still runs other specs. I guess I'll have to run these more frequently [soon](https://github.com/basiszwo/openproject/commit/51a95fefc80615393a541d8548157c7619bb12c8)